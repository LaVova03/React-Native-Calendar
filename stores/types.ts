import { EventFormData } from "@/types/types";

export interface StoreState {
  isModal: {
    form: boolean;
    edit: boolean;
    confirm: boolean;
  };
  data: EventFormData[];
  setData: (events: EventFormData[]) => void;
  setIsModal: (key: keyof StoreState["isModal"], value: boolean) => void;
}
