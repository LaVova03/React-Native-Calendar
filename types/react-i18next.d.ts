import "react-i18next";

declare module "react-i18next" {
  interface Resources {
    en: {
      translation: typeof import("./locales/en.json");
    };
    uk: {
      translation: typeof import("./locales/uk.json");
    };
  }
}
