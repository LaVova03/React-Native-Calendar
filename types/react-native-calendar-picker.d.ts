declare module "react-native-calendar-picker" {
  import * as React from "react";
  import { ViewStyle, TextStyle } from "react-native";

  export interface CalendarPickerProps {
    onDateChange: (date: any) => void;
    selectedDate?: string | Date;
    todayBackgroundColor?: string;
    selectedDayColor?: string;
    selectedDayTextColor?: string;
    headerWrapperStyle?: ViewStyle;
    headerStyle?: ViewStyle;
    monthTitleStyle?: TextStyle;
    dateTextStyle?: TextStyle;
    customDatesStyles?: Array<{
      date: Date;
      style: ViewStyle;
      textStyle: TextStyle;
    }>;
    previousComponent?: React.ReactNode;
    nextComponent?: React.ReactNode;
  }

  export default class CalendarPicker extends React.Component<CalendarPickerProps> {}
}
