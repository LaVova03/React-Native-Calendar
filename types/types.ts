import { Moment } from "moment";
import React from "react";

export interface EventFormData {
  eventName: string;
  startDate: string;
  startTime: string;
  repeat: string;
  id?: number;
}

export interface EventCardProps {
  event: EventFormData;
  setIdEvent?: (value: number | null) => void;
  setPutevent?: (value: EventFormData) => void;
}

export interface FormProps {
  selectedDate?: Moment | null;
  events?: Event[];
  setEvents?: React.Dispatch<React.SetStateAction<Event[]>>;
  editEvent?: EventFormData | null;
}

export interface SelectProps {
  setFormData: React.Dispatch<React.SetStateAction<EventFormData>>;
  formData: EventFormData;
}

export interface ModalFormProps {
  children: React.ReactNode;
}

export interface ConfirmProps {
  setConfirmDelete: React.Dispatch<React.SetStateAction<boolean>>;
}
