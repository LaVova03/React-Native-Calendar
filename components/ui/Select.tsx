import React, { useState } from "react";
import {
  View,
  TextInput,
  Text,
  FlatList,
  TouchableOpacity,
  StyleSheet,
} from "react-native";
import Icon from "react-native-vector-icons/MaterialIcons";
import { SelectProps } from "@/types/types";

export default function Select({ formData, setFormData }: SelectProps) {
  const [inputValue, setInputValue] = useState<string>("");
  const [showDropdown, setShowDropdown] = useState<boolean>(false);

  const options = ["Weekly", "Bi-weekly", "Monthly"];

  return (
    <View style={styles.container}>
      <Text style={styles.text}>Repeat</Text>
      <View style={styles.inputContainer}>
        <TextInput
          style={styles.input}
          value={formData.repeat || inputValue}
          onFocus={() => setShowDropdown(true)} // Показываем список при фокусе на инпуте
          editable={false} // Отключаем возможность ручного ввода
        />
        <TouchableOpacity onPress={() => setShowDropdown(!showDropdown)}>
          <Icon
            name={showDropdown ? "keyboard-arrow-up" : "keyboard-arrow-down"}
            size={24}
            color="gray"
          />
        </TouchableOpacity>
      </View>

      {showDropdown && (
        <FlatList
          contentContainerStyle={styles.dropdownContainer}
          data={options}
          keyExtractor={(item) => item}
          renderItem={({ item }) => (
            <TouchableOpacity
              style={styles.dropdownItem}
              onPress={() => {
                setFormData((prev) => ({
                  ...prev,
                  repeat: item,
                }));
                setInputValue(item); // Обновляем инпут выбранным значением
                setShowDropdown(false); // Закрываем список
              }}
            >
              <Text>{item}</Text>
            </TouchableOpacity>
          )}
        />
      )}
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "flex-end",
    alignItems: "center",
    padding: 0,
  },
  text: {
    fontSize: 18,
    marginBottom: 10,
  },
  inputContainer: {
    flexDirection: "row",
    alignItems: "flex-end",
    width: "100%",
    backgroundColor: "#f0f0f0",
    borderRadius: 5,
    padding: 10,
    justifyContent: "space-between",
  },
  input: {
    flex: 1,
    padding: 0,
    backgroundColor: "transparent",
  },
  dropdownContainer: {
    width: "100%",
  },
  dropdownItem: {
    padding: 10,
    backgroundColor: "#f0f0f0",
    borderWidth: 1,
    borderColor: "#ddd",
    width: 100,
  },
});
