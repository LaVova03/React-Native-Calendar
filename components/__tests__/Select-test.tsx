import * as React from "react";
import renderer from "react-test-renderer";
import Select from "@/components/ui/Select";
import { EventFormData } from "@/types/types";

jest.mock("react-native-vector-icons/MaterialIcons", () => "Icon");

const mockSetFormData = jest.fn();

const mockFormData: EventFormData = {
  eventName: "Test Event",
  startDate: "2024-02-25",
  startTime: "12:00",
  repeat: "",
};

it("renders Select component correctly", () => {
  const tree = renderer
    .create(<Select formData={mockFormData} setFormData={mockSetFormData} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
