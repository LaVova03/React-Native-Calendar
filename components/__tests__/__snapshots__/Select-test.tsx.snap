// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Select component correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "flex-end",
      "padding": 0,
    }
  }
>
  <Text
    style={
      {
        "fontSize": 18,
        "marginBottom": 10,
      }
    }
  >
    Repeat
  </Text>
  <View
    style={
      {
        "alignItems": "flex-end",
        "backgroundColor": "#f0f0f0",
        "borderRadius": 5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 10,
        "width": "100%",
      }
    }
  >
    <TextInput
      editable={false}
      onFocus={[Function]}
      style={
        {
          "backgroundColor": "transparent",
          "flex": 1,
          "padding": 0,
        }
      }
      value=""
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Icon
        color="gray"
        name="keyboard-arrow-down"
        size={24}
      />
    </View>
  </View>
</View>
`;
